interfaces/builtin/io_ports_control.go

```diff
@@ -0,0 +1,53 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const ioPortsControlConnectedPlugAppArmor = `
+# Description: Allow write access to all I/O ports.
+# See 'man 4 mem' for details.
+
+capability sys_rawio, # required by iopl
+
+/dev/ports rw,
```

Comment:
I didn't think of this before, but you are going to want to also udev tag this since I suspect this interface is going to be used alongside things like i2c, iio, serial-port, etc. See https://github.com/snapcore/snapd/pull/2459#discussion_r92240828. I would be ok with merging this as is if 2.20 is a hard requirement and then fixing this for 2.21.

See https://github.com/snapcore/snapd/pull/2459/files#diff-0fc98d9f1337c18cf172845c6cd2a7b5R119 on how to do this.